impl/src/main/kotlin/com/android/declarative/internal/SettingsDeclarativePlugin.kt

```diff
@@ -135,24 +162,12 @@ class SettingsDeclarativePlugin @Inject constructor(
                     .forEach("focus", listOfProjects::add)
                 return listOfProjects.toList()
             }
-            return if (parsedDecl.contains("focus")) {
-                parsedDecl.safeGetString("focus").split(",")
-            } else {
-                val listOfProjects = mutableListOf<String>()
-                val regex = Regex("(:[^:]*):(.*)")
-                settings.gradle.startParameter.taskNames.forEach {
-                    if (regex.matches(it)) {
-                        regex.matchEntire(it)?.groups?.get(1)?.value?.let { projectName ->
-                            println("Focusing on $projectName from task name $it")
-                            listOfProjects.add(projectName)
-                        }
-                    }
-                }
-                listOfProjects.toList()
+            if (parsedDecl.contains("focus")) {
+                return parsedDecl.safeGetString("focus").split(",")
```

Comment:
Minor: maybe we shouldn't use `contains` check and `safeGetString` here but instead `parsedDecl.getString("focus")?.let{return it.split(",")}`

Reply:
yes good point.